src/lolex.js

```diff
@@ -258,6 +258,42 @@
         }
     }
 
+    function timerType(timer) {
+        if (timer.immediate) {
+            return "immediate";
+        } else if (typeof timer.interval !== "undefined") {
+            return "interval";
+        } else {
+            return "timeout";
+        }
+    }
+
+    function clearTimer(clock, timerId, ttype) {
+        if (!timerId) {
+            // null appears to be allowed in most browsers, and appears to be
+            // relied upon by some libraries, like Bootstrap carousel
+            return;
+        }
+
+        if (!clock.timers) {
+            clock.timers = [];
```

Comment:
I agree, but I did not change that code. I moved that entire function body from its old place in clock.clearTimeout() and only added the last few lines.  
That thing with setting clock.timers to [] was so specific that I thought it must have a function and I didn't want to mess with it. Want me to change it anyway?